src/cdk-experimental/scrolling/virtual-scroll-viewport.spec.ts

```diff
@@ -1,46 +1,394 @@
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        imports: [ScrollingModule],
+        declarations: [FixedVirtualScroll],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(FixedVirtualScroll);
+      testComponent = fixture.componentInstance;
+      viewport = testComponent.viewport;
+    });
+
+    it('should sanitize transform inputs', fakeAsync(() => {
+      finishInit(fixture);
+      viewport.orientation = 'arbitrary string as orientation' as any;
+      viewport.setRenderedContentOffset(
+          'arbitrary string as offset' as any, 'arbitrary string as to' as any);
+      fixture.detectChanges();
+      flush();
+
+      expect((viewport._renderedContentTransform as any).changingThisBreaksApplicationSecurity)
+          .toEqual('translateY(NaNpx)');
+    }));
+
+    it('should render initial state', fakeAsync(() => {
+      finishInit(fixture);
+
+      const contentWrapper =
+          viewport.elementRef.nativeElement.querySelector('.cdk-virtual-scroll-content-wrapper');
+      expect(contentWrapper.children.length)
+          .toBe(4, 'should render 4 50px items to fill 200px space');
+    }));
+
+    it('should get the data length', fakeAsync(() => {
+      finishInit(fixture);
+
+      expect(viewport.getDataLength()).toBe(testComponent.items.length);
+    }));
+
+    it('should get the viewport size', fakeAsync(() => {
+      finishInit(fixture);
+
+      expect(viewport.getViewportSize()).toBe(testComponent.viewportSize);
+    }));
+
+    it('should get the rendered range', fakeAsync(() => {
+      finishInit(fixture);
+
+      expect(viewport.getRenderedRange())
+          .toEqual({start: 0, end: 4}, 'should render the first 4 50px items to fill 200px space');
+    }));
+
+    it('should get the rendered content offset', fakeAsync(() => {
+      finishInit(fixture);
+      triggerScroll(viewport, testComponent.itemSize + 5);  // Scroll 5px into the second item.
+
+      expect(viewport.getOffsetToRenderedContentStart()).toBe(testComponent.itemSize,
+          'should have 50px offset since first 50px item is not rendered');
+    }));
+
+    it('should get the scroll offset', fakeAsync(() => {
+      finishInit(fixture);
+      triggerScroll(viewport, testComponent.itemSize + 5);  // Scroll 5px into the second item.
+
+      expect(viewport.measureScrollOffset()).toBe(testComponent.itemSize + 5);
+    }));
+
+    it('should get the rendered content size', fakeAsync(() => {
+      finishInit(fixture);
+
+      expect(viewport.measureRenderedContentSize())
+          .toBe(testComponent.viewportSize,
+              'should render 4 50px items with combined size of 200px to fill 200px space');
+    }));
+
+    it('should measure range size', fakeAsync(() => {
+      finishInit(fixture);
+
+      expect(viewport.measureRangeSize({start: 1, end: 3}))
+          .toBe(testComponent.itemSize * 2, 'combined size of 2 50px items should be 100px');
```

Comment:
It would also fail if we hard-coded the number and then changed the item size. I would prefer to use values like this over raw numbers when possible, otherwise you have no idea what the numbers are supposed to be when you come back to it in the future. (I had this experience when I refactored the select to work inside form-field. The test file was full of numbers that meant nothing to me).